libs/agno/agno/workflow/v2/workflow.py

```diff
@@ -1116,6 +1137,281 @@ def _update_workflow_session_state(self):
+            run_id=self.run_id,
+            session_id=self.session_id,
+            workflow_id=self.workflow_id,
+            workflow_name=self.name,
+            created_at=int(datetime.now().timestamp()),
+            status=RunStatus.pending,
+            content="Workflow execution started in background",
+        )
+
+        # Store PENDING response immediately
+        if self.workflow_session:
+            self.workflow_session.add_run(workflow_run_response)
+        self.write_to_storage()
+
+        # Prepare execution input
+        inputs = WorkflowExecutionInput(
+            message=message,
+            additional_data=additional_data,
+            audio=audio,  # type: ignore
+            images=images,  # type: ignore
+            videos=videos,  # type: ignore
+        )
+
+        self.update_agents_and_teams_session_info()
+
+        async def execute_workflow_background():
+            """Direct execution wrapper with minimal overhead"""
+            try:
+                # Quick status update to RUNNING
+                self.update_background_status(self.run_id or "", RunStatus.running)
+
+                loop = asyncio.get_event_loop()
+                await loop.run_in_executor(
+                    None,
+                    lambda: self._execute(
+                        execution_input=inputs, workflow_run_response=workflow_run_response, **kwargs
+                    ),
+                )
+
+                # result IS workflow_run_response (same object, modified by _execute)
+                # Quick status update to database
+                self.update_background_status(self.run_id or "", workflow_run_response.status)
+
+                log_debug(f"Background execution completed with status: {workflow_run_response.status}")
+                log_debug(
+                    f"Content length: {len(str(workflow_run_response.content)) if workflow_run_response.content else 0}"
+                )
+
+            except Exception as e:
+                logger.error(f"Background workflow execution failed: {e}")
+                workflow_run_response.status = RunStatus.error
+                workflow_run_response.content = f"Background execution failed: {str(e)}"
+                self.update_background_status(self.run_id or "", RunStatus.error)
+
+        # Create and start asyncio task
+        try:
+            loop = asyncio.get_running_loop()
+            loop.create_task(execute_workflow_background())
+        except RuntimeError:
+            # No event loop, use threading fallback
+            import threading
+
+            def run_in_thread():
+                new_loop = asyncio.new_event_loop()
+                asyncio.set_event_loop(new_loop)
+                try:
+                    new_loop.run_until_complete(execute_workflow_background())
+                finally:
+                    new_loop.close()
+
+            thread = threading.Thread(target=run_in_thread, daemon=True)
```

Comment:
Threading will be blocking though?

Reply:
just noticed that for the sync code (using .run) we cannot really use create_task as it requires a running event loop (i.e we need to be inside an async context)
so even for sync bg execution, in the cookbook example we'd have to do something like this...
is that an acceptable DX?

```py
async def main():
    # Start background execution
    bg_response = content_creation_workflow.run(
        message="AI trends in 2024", background=True
    )

    ... other logic

if __name__ == "__main__":
    asyncio.run(main())
```
right now i was falling back to threading for the sync case.